### ISSUE TITLE
Maybe false_positive in cast_possible_truncation?

### ISSUE THREAD

**Issue by darnir** [rust-lang/rust-clippy#2746]
When the code looks like this:

```Rust
let a: f64 = 145.98;
let b = a.ceil() as u64;
```

Then maybe, it makes sense to disable the `cast_possible_truncation` lint on it?
This is because methods like `ceil`, `round`, `floor` and `trunc` will all return an integer as a floating point expression. So truncating to an integer causes no loss of data, and moreover, this is exactly what the programmer wanted. Since they explicitly called a method which returns a integral value represented as a floating point.

Or am I missing something here?

**Comment by mikerite** [comment]
This isn't a bug. If the float is too large it will be truncated. 

For example
```
fn main() {
    let a: f64 = 1E50;
    let b = a.ceil();
    let c = b as u64;
    println!("b: {}", b);
    println!("c: {}", c);
}
```

outputs 
```
b: 100000000000000000000000000000000000000000000000000
c: 0
```

**Comment by darnir** [comment]
You're right. This makes sense. I was purely thinking of it in terms of the fact than the mantissa of a `f64` is smaller than 64-bits and hence there should be no truncation. But your example shows that it the assumption there is incorrect.
